cmd/snap-confine/snap-confine.c

```diff
@@ -108,9 +260,17 @@ int main(int argc, char **argv)
 	// Use our super-defensive parser to figure out what we've been asked to do.
 	struct sc_error *err = NULL;
 	struct sc_args *args SC_CLEANUP(sc_cleanup_args) = NULL;
+	sc_preservation_context SC_CLEANUP(sc_cleanup_preservation_context)
+	    prsv_ctx = {
+	.orig_umask = 0,.orig_cwd_fd = -1};
 	args = sc_nonfatal_parse_args(&argc, &argv, &err);
 	sc_die_on_error(err);
 
+	// Remember certain properties of the process that are clobbered by
+	// snap-confine during execution. Those are restored just before calling
+	// execv.
+	sc_remember_preservation_context(&prsv_ctx);
```

Comment:
The comment and the name as commented earlier, do not really mention that the properties of the process are clobbered inside too.

Reply:
I agree, I will continue in the older PR